src/actions/mod.rs

```diff
@@ -451,26 +452,38 @@ impl ActionHandler {
                 return;
             }
         };
-        let config = self.fmt_config.lock().unwrap();
-        let mut config = config.get_rustfmt_config().clone();
+
+        let range_whole_file = ls_util::range_from_vfs_file(&self.vfs, path);
+        let mut config = self.fmt_config.lock().unwrap().get_rustfmt_config().clone();
         if !config.was_set().hard_tabs() {
             config.set().hard_tabs(!opts.insert_spaces);
         }
         if !config.was_set().tab_spaces() {
             config.set().tab_spaces(opts.tab_size as usize);
         }
 
+        if let Some(r) = selection {
+            let range_of_rls = ls_util::range_to_rls(r).one_indexed();
+            let range = RustfmtRange::new(range_of_rls.row_start.0 as usize, range_of_rls.row_end.0 as usize);
+            let mut ranges = HashMap::new();
+            ranges.insert("stdin".to_owned(), vec![range]);
+            let file_lines = FileLines::from_ranges(ranges);
+            config.set().file_lines(file_lines);
+        };
```

Comment:
Fix up to create `FileLines` directly instead of using `format!`.
At present, it destroys the build, until [this patch](https://github.com/rust-lang-nursery/rustfmt/pull/1735) is released 🙏